syntax = "proto3";
package tg;

enum ELanguage {
    LN_UNDEFINED = 0;
    LN_RU = 1;
    LN_EN = 2;

    LN_OTHER = 3;
}

enum ECategory {
    NC_UNDEFINED = 0;
    NC_ANY = 1;
    NC_SOCIETY = 2;
    NC_ECONOMY = 3;
    NC_TECHNOLOGY = 4;
    NC_SPORTS = 5;
    NC_ENTERTAINMENT = 6;
    NC_SCIENCE = 7;
    NC_OTHER = 8;
    NC_NOT_NEWS = 9;
}

enum EEmbedderType {
    ET_UNDEFINED = 0;
    ET_FASTTEXT = 1;
    ET_TORCH = 2;
    ET_TFIDF = 3;
}

enum EEmbeddingKey {
    EK_UNDEFINED = 0;
    EK_FASTTEXT_CLASSIC = 1;
    EK_FASTTEXT_TITLE = 2;
    EK_FASTTEXT_TEXT = 3;
    EK_TFIDF = 4;
}

enum EAggregationMode {
    AM_UNDEFINED = 0;
    AM_AVG = 1;
    AM_MAX = 2;
    AM_MIN = 3;
    AM_MATRIX = 4;
}

enum EEmbedderField {
    EF_UNDEFINED = 0;
    EF_TITLE = 1;
    EF_TEXT = 2;
    EF_ALL = 3;
}

enum EInputFormat {
    IF_UNDEFINED = 0;
    IF_HTML = 1;
    IF_JSON = 2;
    IF_JSONL = 3;
}
